src/main/java/pro/quicksense/modules/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,216 @@
+package pro.quicksense.modules.service.impl;
+
+
+import cn.hutool.core.util.ObjectUtil;
+import com.baomidou.mybatisplus.core.conditions.query.LambdaQueryWrapper;
+import com.baomidou.mybatisplus.core.toolkit.StringUtils;
+import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pro.quicksense.modules.common.CommonConstant;
+import pro.quicksense.modules.entity.EmailLoginRequest;
+import pro.quicksense.modules.entity.LoginRequest;
+import pro.quicksense.modules.entity.User;
+import pro.quicksense.modules.mapper.UserMapper;
+import pro.quicksense.modules.service.UserService;
+import pro.quicksense.modules.util.CodeUtil;
+import pro.quicksense.modules.util.EmailUtil;
+import org.springframework.http.ResponseEntity;
+import pro.quicksense.modules.util.JwtInterceptor;
+
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class UserServiceImpl extends ServiceImpl<UserMapper, User> implements UserService {
+
+    private final EmailUtil emailUtil;
+    private final CodeUtil codeUtil;
+    private final JwtInterceptor jwtInterceptor;
+
+
+    @Transactional
```

Comment:
Do we need `@Transactional` annotation here? If `this.save(user)` fails, it means that the data has not yet been inserted into the database.